tests/test_data_handling.py

```diff
@@ -10,7 +9,8 @@
 
 
 @pytest.mark.django_db
-def test_data_handler(form_entry, location, study, form_element, mocker):
+def test_data_handler(form_entry, location, study, form_element,
+                      form_element_help_text, mocker):
```

Comment:
I see what you're saying about this test being long. In this case I think it's fine, since what you're trying to do is save one type of object and make sure that a bunch of different attributes got saved correctly. If the test were saving different types of objects and testing their attributes, it would make sense to me to split it out into multiple tests, but in this case I think it's better to sacrifice brevity of the test in order to minimize database access and scope the test to one specific unit of functionality.

In general I tend to be less strict about code quality concerns (like DRY and keeping functions short) when writing tests. Tests are often verbose, which I think is OK since they're not really "methods" in a traditional sense -- they're usually more declarative than imperative, and I typically try to write them keeping in mind that the person reading them will most likely be skimming to try to debug a failure, not reuse functionality.

Reply:
Very helpful, thanks for explaining!